app/components/protected-monument-select.js

```diff
@@ -0,0 +1,22 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import fetch from 'fetch';
+import { restartableTask } from 'ember-concurrency';
+
+export default class AddressRegisterSelectorComponent extends Component {
+  @service store;
+
+  @restartableTask
+  *search(searchData) {
```

Comment:
Suggestion to add the debounce?  
`yield timeout(DEBOUNCE_MS);`

Reply:
yes totally ! I added it thx :)